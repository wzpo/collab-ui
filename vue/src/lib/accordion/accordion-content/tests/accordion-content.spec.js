import { shallowMount } from '@vue/test-utils';
import AccordionContent from '../index.vue';

describe('AccordionContent', () => {
  it('should match snapshot', () => {
    const wrapper = shallowMount(AccordionContent);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should render one AccordionContent', () => {
    const container = shallowMount(AccordionContent);

    expect(container.findAll('.md-accordion__content').length).toEqual(1);
  });

  it('should pass ClassName Prop', () => {
    const container = shallowMount(AccordionContent, {
      attrs: {
        class: 'testClass'
      }
    });

    expect(container.find('.testClass').exists()).toEqual(true);
  });

  it('should render children', () => {
    const container = shallowMount(AccordionContent, {
      slots: {
        default: `<div class="testingforTC" />`
      }
    });

    expect(container.findAll('.testingforTC').length).toEqual(1);
  });
});

