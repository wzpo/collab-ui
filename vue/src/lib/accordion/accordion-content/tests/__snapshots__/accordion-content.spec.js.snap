// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionContent should match snapshot 1`] = `<div class="md-accordion__content"></div>`;
